src/main/java/Duke.java

```diff
@@ -1,10 +1,214 @@
+import duke.tasks.*;
+import duke.exceptions.*;
+import duke.files.FileManager;
+
+import java.io.IOException;
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
+    public static void main(String[] args) throws Exception {
+
+        // Text declarations and initial greeting
+        final String BARRIER = "____________________________________________________________";
+        final String LOGO = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+        final String GREETING = BARRIER + "\n\nAhoy there! I be\n" + LOGO + "\nWhat can I do for ye?"+
+        "\nFor a list of commands, type 'help' me hearties!\n" + BARRIER + "\n";
+        final String EXIT = BARRIER + "\n\nFarewell! Hope to see ye again soon, ye scallywag!\n" + BARRIER;
+        System.out.println(GREETING);
+
+        // Scanner declaration and initial input
+        final Scanner CONSOLE = new Scanner(System.in);
+        String input = CONSOLE.nextLine();
+
+        // Scan data.txt for saved tasks
+        try {
+            FileManager.populateTaskList();
+        } catch (FileNotFoundException e) {
+            System.out.println("Argh! There was an issue fetching yer saved tasks! Ye must be in the 'java' directory!");
+        }
+
+        // Read input such that it is not "bye"
+        while(!input.equals("bye")) {
+
+            // Split user input to check for dynamism
+            String originalInput = input;
+            String[] splitInput = input.split(" ");
+            input = splitInput[0];
+            int numTasks = Task.getTasksArray().size();
+
+            // Switch to check input
+            switch(input){
```

Comment:
For your case blocks, consider abstracting them out to separate functions so as to reduce deeply nested code blocks to increase readability.